quiche/src/recovery/mod.rs

```diff
@@ -461,6 +465,76 @@ impl QlogMetrics {
     }
 }
 
+/// When the pacer thinks is a good time to release the next packet
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ReleaseTime {
+    Immediate,
+    At(Instant),
+}
+
+/// When the next packet should be release and if it can be part of a burst
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct ReleaseDecision {
+    time: ReleaseTime,
+    allow_burst: bool,
+}
+
+impl ReleaseTime {
+    /// Add the specific delay to the current time
+    #[allow(dead_code)]
+    fn inc(&mut self, delay: Duration) {
+        match self {
+            ReleaseTime::Immediate => {},
+            ReleaseTime::At(time) => *time += delay,
+        }
+    }
+
+    /// Set the time to the later of two times
+    #[allow(dead_code)]
+    fn set_max(&mut self, other: Instant) {
+        match self {
+            ReleaseTime::Immediate => *self = ReleaseTime::At(other),
+            ReleaseTime::At(time) => *self = ReleaseTime::At(other.max(*time)),
+        }
+    }
+}
+
+impl ReleaseDecision {
+    pub(crate) const EQUAL_THRESHOLD: Duration = Duration::from_micros(35);
+
+    /// Get the [`Instant`] the next packet should be released. It will never be
+    /// in the past.
+    #[allow(dead_code)]
+    #[inline]
+    pub fn time(&self, now: Instant) -> Option<Instant> {
```

Comment:
post: should this simple return ReleaseDecision. We re-interpret `Immediate` as `None` which is a bit confusing.

post: rename function to `release_time`?